packages/fungible-token/bridge-fungible-token/src/data_structures.sw

```diff
@@ -15,7 +15,7 @@ impl MessageData {
     ///
     /// any data beyond 160 bytes means deposit is meant for a contract.
     /// if data is > 161 bytes, then we also need to call process_message on the destination contract.
-    pub fn parse(msg_idx: u8) -> Self {
+    pub fn parse(msg_idx: u64) -> Self {
```

Comment:
why did we change this to u64?

Reply:
ah this is likely because sway doesn't allow for implicit integer casting now? Since I don't think the stdlib changed here. The strange thing to me is that it seems like we should be able to safely upcast ints without any concerns for precision.

cc @IGI-111

Reply:
Yeah we have safe upcasts, though they have to be explicitly called.

I'm not sure if this should be fixed here or on the library side however.